sonaranalyzer-dotnet/rspec/vbnet/S4823_vb.net.html

```diff
@@ -0,0 +1,26 @@
+<pre>
+Module Program
+    Sub Main(args As String()) ' Questionable as there is a reference to "args" in the procedure.
+        Console.WriteLine(args(0))
```

Comment:
The example might be better if the args were being used in a more obviously unsafe way e.g. passing them to another method in class. Passing them to _Console.WriteLine_ doesn't seem to be unsafe so it's easy to overlook the value of the rule i.e. the examples should maximize the risk to highlight the value of the rules. @nicolas-harraudeau-sonarsource, what do you think?

Reply:
@duncanp-sonar This is of course open to discussion but I would keep this topic for later. In my experience every person has a different point of view on what examples should represent. Some see it as specification of what the rule highlight, others see it as a minimalistic example showing realistic code.
For these rules I chose to show what Hotspot highlight. The point of a Security Hotspot is to ask for a review of sensitive code. It does not pinpoint vulnerabilities. Showing only vulnerabilities could make the person think that the rule is smarter than it actually is.
It is the role of the textual description to highlight the risks and explain how to avoid them. There are also links to CERT, OWASP and SANS-TOP-25 showing vulnerability examples.

Reply:
This is from the unit tests, the idea was to check if we catch more than simple identifier syntax (this is ElementAccessExpressionSyntax)